core/network/address_options.go

```diff
@@ -52,8 +61,14 @@ func WithCIDR(cidr string) func(AddressMutator) {
 
 // WithSecondary returns a functional option that can be used to
 // indicate that an address is not the primary for its NIC.
-func WithSecondary() func(mutator AddressMutator) {
+func WithSecondary() func(AddressMutator) {
 	return func(a AddressMutator) {
 		a.SetSecondary()
 	}
 }
+
+func WithConfigType(configType AddressConfigType) func(AddressMutator) {
```

Comment:
I added this option early in the work on this patch, before refining how much I was biting off.
I will add tests for it in the next bit of work; it is not yet used.